jpc/src/lib.rs

```diff
@@ -2413,12 +2647,27 @@ impl ContiguousCodestream {
                         header.comment_marker_segments.push(comment_marker_segment);
                     }
 
+                    // CAP (Optional)
+                    // TODO: in strict mode, ensure this is the first marker segment after SIZ
```

Comment:
To facilitate the validation we could store the order of the markers in the main header, this could also improve jpxml to ensure we output it in order of where we see it. Thoughts?